apps/indexer/lib/indexer/fetcher/token_instance/metadata_retriever.ex

```diff
@@ -20,17 +20,46 @@ defmodule Indexer.Fetcher.TokenInstance.MetadataRetriever do
 
   @ignored_hosts ["localhost", "127.0.0.1", "0.0.0.0", "", nil]
 
-  defp ipfs_link do
-    link =
+  @spec ipfs_link(uid :: any()) :: String.t()
+  defp ipfs_link(uid) do
+    base_url =
       :indexer
-      |> Application.get_env(:ipfs_gateway_url)
+      |> Application.get_env(:ipfs)
+      |> Keyword.get(:gateway_url)
       |> String.trim_trailing("/")
 
-    link <> "/"
+    url = base_url <> "/" <> uid
+
+    ipfs_params = Application.get_env(:indexer, :ipfs)
+
+    if ipfs_params[:gateway_url_param_location] == :query do
+      gateway_url_param_key = ipfs_params[:gateway_url_param_key]
+      gateway_url_param_value = ipfs_params[:gateway_url_param_value]
+
+      if gateway_url_param_key && gateway_url_param_value do
+        url <> "?#{gateway_url_param_key}=#{gateway_url_param_value}"
```

Comment:
Seems like there should be smth like
```suggestion
        url <> "?#{gateway_url_param_key}=#{gateway_url_param_value}"
      else
        url
```